core/src/test/java/org/eclipse/hono/util/HostnamesTest.java

```diff
@@ -11,21 +11,21 @@
  * SPDX-License-Identifier: EPL-2.0
  *******************************************************************************/
 
-package org.eclipse.hono.adapter.mqtt;
+package org.eclipse.hono.util;
 
-import org.eclipse.hono.service.metric.DropwizardBasedMetrics;
-import org.springframework.stereotype.Component;
+import org.junit.Assert;
+import org.junit.Test;
 
 /**
- * Metrics for the MQTT adapter.
+ * Unit tests for {@link Hostnames}.
  */
-@Component
-public class DropwizardBasedMqttAdapterMetrics extends DropwizardBasedMetrics implements MqttAdapterMetrics {
+public class HostnamesTest {
 
-    private static final String SERVICE_PREFIX = "hono.mqtt";
-
-    @Override
-    protected String getScope() {
-        return SERVICE_PREFIX;
+    /**
+     * The result must never be {@code null}.
+     */
+    @Test
+    public void testNonNull() {
+        Assert.assertNotNull(Hostnames.getHostname());
```

Comment:
this is not a reproducible test because it doesn't test the specific ways for determining the hostname, does it?

Reply:
Well this test only tests if "not null" is returned, and that should be reproducible.

Reply:
You're right in that it always succeeds because the method always returns a non-null value. But that is a no-brainer, isn't it? A reasonable test should actually check if it returns a value based on the different ways the getHostname method tries to determine the hostname, shouldn't it?

Reply:
>  But that is a no-brainer, isn't it?

At the moment yes. But the actual lookup of the value is not. And if someone changes the logic of that method, accidentally returning `null`, this test would catch that.

Yes, that unit test could have additional methods. But I am not sure how feasible it will be creating them. Modifying the environment variable HOSTNAME inside a unit test might be bad idea.